requirements.txt

```diff
@@ -8,6 +8,7 @@ mapclassify
 matplotlib
 notebook
 numpy
+ruff
```

Comment:
I think we should be going all in on one package manager and virtual environment manager. Whether that be pip/venv, Pipenv, or Poetry I don't care, but we should have an opinionated method of running this. If we're set on a `pyproject.toml` I would move the dependencies there to avoid any confusion and future cases of "it works on my machine but not yours". I would also move the makefile scripts into scripts in `pyproject.toml` as well, as having `make` as a dependency can cause some friction to the end user.

Reply:
By default, the opinion on the package manager/virtual environment manager is pip/venv. The current instructions are for pip/venv.

The current instructions also cover the `pyproject.toml` file as well. Currently, the documented way to install dependencies is `pip install -r requirements.txt`, which will install all of the requirements necessary as well as the source code in `src` as an editable install.

Reply:
Has anyone solved "it works on my machine but not yours"?
Can we implement a Docker container?

Reply:
Interesting callout, but keep in mind that the end users of this will (probably) be American Red Cross workers running this on their local laptops. IMO, running in a docker container just adds another layer of complexity and friction to running these scripts. A virtual environment fixes issues of python and dependency differences, and I feel like that's enough friction to running this. Running docker (essentially a VM) also imposes a higher degree of compute requirements.

Reply:
I'm coming from a background in web apps... So I thought we were developing locally but then running on AWS or Heroku.

I agree with you: Docker adds more complexity if the final product will be running on the users' laptops.

So with the virtual environment, it should run on any computer? Or will there be any initial set up required from each user?

Reply:
they'll need python installed. then the virtual environment keeps the dependencies for this project separate from what they might install for a different python project. installing the required python libraries is just 1 command. we'll have [the setup documented in the readme](https://github.com/AmericanRedCross/street-view-green-view/?tab=readme-ov-file#0-setup).